packages/experiment-tag/src/experiment.ts

```diff
@@ -186,6 +189,59 @@ const revertMutations = () => {
   }
 };
 
+const inject = (js: string, html, utils, id) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  eval(js, html, utils, id);
+};
+
+const handleInject = (action, key: string, variant: Variant) => {
+  const globalScope = getGlobalScope();
+  if (!globalScope) {
+    return;
+  }
+  const urlExactMatch = variant?.metadata?.['urlMatch'] as string[];
+  const currentUrl = urlWithoutParamsAndAnchor(globalScope.location.href);
+  if (matchesUrl(urlExactMatch, currentUrl)) {
```

Comment:
@tyiuhc I think this is going to conflict with your PR?

Reply:
Yes, with page targeting, we would be able to forgo the matchUrl condition. We could merge your PR first, then I can update the logic and tests with page targeting.